analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.CSharp8.cs

```diff
@@ -0,0 +1,209 @@
+
+    public void Deconstruct(out string name, out string state) =>
+        (name, state) = (Name, State);
+}
+
+public class Person
+{
+    public string Name { get; }
+
+    public Address Address { get; }
+
+    public void Deconstruct(out string name, out Address address) =>
+        (name, address) = (Name, Address);
+}
+
+public class SwitchExpressions
+{
+    public void OnlyDiscardBranch_Noncompliant(string s, bool b)
+    {
+        var result = b switch
+        {
+            _ => s.ToString() // FIXME Non-compliant
+        };
+    }
+
+    public void MultipleBranches_Noncompliant(string s, int val)
+    {
+        var result = val switch
+        {
+            1 => "a",
+            2 => s.ToString(), // FIXME Non-compliant
+            _ => "b"
+        };
+    }
+
+    public void Nested_Noncompliant(string s, int val, bool condition)
+    {
+        var result = val switch
+        {
+            1 => "a",
+            2 => condition switch
+            {
+                _ => s.ToString() // FIXME Non-compliant
+            },
+            _ => "b"
+        };
+    }
+
+    public void MultipleBranches_HandleNull(string s, int val)
+    {
+        var result = s switch
+        {
+            null => s.ToString(), // FIXME Non-compliant
+            _ => s.ToString() // Compliant as null was already handled
+        };
+    }
+
+    public void MultipleBranches_Compliant(string s, int val)
+    {
+        var result = val switch
+        {
+            1 => "a",
+            2 => s == null ? string.Empty : s.ToString(),
+            _ => "b"
+        };
+    }
+
+    public string MultipleBranches_PropertyPattern(Address address, string s)
+    {
+        return address switch
+        {
+            { State: "WA" } addr => s.ToString(), // FIXME Non-compliant
+            _ => string.Empty
+        };
+    }
+
+    public string MultipleBranches_PropertyPattern_FP(string s)
+    {
+        return s switch
+        {
+            { Length: 5 } => s.ToString(), // FIXME Non-compliant - FP we know that the length is 5 so the string cannot be null
+            _ => string.Empty
+        };
+    }
+
+    public string MultipleBranches_RecursivePattern(Person person, string s)
+    {
+        return person switch
+        {
+            { Address: { State: "WA" } } pers => s.ToString(), // FIXME Non-compliant
+            _ => string.Empty
+        };
+    }
+
+    public string MultipleBranches_TuplePattern(Address address, string s)
+    {
+        return address switch
+        {
+            var (name, state) => s.ToString(), // Compliant - FN
+            _ => string.Empty
+        };
+    }
+
+    public string MultipleBranches_WhenClause(Address address, string s)
+    {
+        return address switch
+        {
+            Address addr when addr.Name.Length > 0 => s.ToString(), // FIXME Non-compliant
+            Address addr when addr.Name.Length == s.Length => string.Empty, // FIXME Non-compliant
```

Comment:
Aligning short blocks on the same tab distance makes it easier to read and understand
```suggestion
            Address addr when addr.Name.Length > 0 => s.ToString(),                     // FIXME Non-compliant
            Address addr when addr.Name.Length == s.Length => string.Empty,    // FIXME Non-compliant
```
see this
https://github.com/SonarSource/sonar-dotnet/blob/d73c2cce42f48ca4f91c6003a2186157292bf5be/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CertificateValidationCheck.cs#L139-L146
vs. this counter example
https://github.com/SonarSource/sonar-dotnet/blob/d73c2cce42f48ca4f91c6003a2186157292bf5be/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssignmentInsideSubExpression.cs#L105-L124